ReactNative/ReactTwitter/android/app/src/main/java/com/reacttwitter/widgets/ReactButtonShadowNode.java

```diff
@@ -0,0 +1,140 @@
+package com.reacttwitter.widgets;
+
+import android.graphics.Rect;
+import android.graphics.Typeface;
+import android.graphics.drawable.Drawable;
+import android.support.v4.content.ContextCompat;
+import android.text.BoringLayout;
+import android.text.Layout;
+import android.text.StaticLayout;
+import android.text.TextPaint;
+
+import com.facebook.csslayout.CSSConstants;
+import com.facebook.csslayout.CSSMeasureMode;
+import com.facebook.csslayout.CSSNode;
+import com.facebook.csslayout.MeasureOutput;
+import com.facebook.react.uimanager.LayoutShadowNode;
+import com.facebook.react.uimanager.PixelUtil;
+import com.facebook.react.uimanager.annotations.ReactProp;
+
+import javax.annotation.Nullable;
+
+public class ReactButtonShadowNode extends LayoutShadowNode implements CSSNode.MeasureFunction {
+
+    private static final TextPaint textPaintInstance = new TextPaint();
+
+    static {
+        textPaintInstance.setFlags(TextPaint.ANTI_ALIAS_FLAG);
+        textPaintInstance.setTypeface(Typeface.create(Typeface.DEFAULT, Typeface.BOLD)); // our button has a bold typeface by default
+    }
+
+    private String text;
+    private final Rect padding;
+
+    public ReactButtonShadowNode() {
+        padding = new Rect();
+        setMeasureFunction(this);
+    }
+
+    @Override
+    protected void markUpdated() {
+        super.markUpdated();
+        super.dirty();
+    }
+
+    @ReactProp(name = "text")
+    public void setText(@Nullable String text) {
+        this.text = text;
+        markUpdated();
+    }
+
+    @ReactProp(name = "textSize", defaultInt = 14)
+    public void setTextSize(int textSize) {
+        textPaintInstance.setTextSize(PixelUtil.toPixelFromSP(textSize));
+        markUpdated();
+    }
+
+    @ReactProp(name = "backgroundImage")
+    public void setBackgroundImage(String backgroundImage) {
+        Drawable drawable = getDrawableByName(backgroundImage);
+        if (drawable == null) {
+            padding.set(0, 0, 0, 0);
+            return;
+        }
+        drawable.getPadding(this.padding);
+        markUpdated();
+    }
+
+    private Drawable getDrawableByName(String name) {
+        int resId = getThemedContext().getResources().getIdentifier(name.toLowerCase(), "drawable", getThemedContext().getPackageName());
+        if (resId <= 0) {
```

Comment:
❌  from `getIdentifier()` documentation:
`int The associated resource identifier. Returns 0 if no such resource was found. (0 is not a valid resource ID.)`
So, `-1` for example is a valid id. You should check for 0 only